### DISCUSSION TITLE
Version 4.2!

### DISCUSSION THREAD

**Discussion by mikebrady** [mikebrady/shairport-sync#1679] [Announcements]
Hi everyone. Shairport Sync 4.2 is now available! It contains important bug fixes and some small enhancements that should make it a lot more reliable, for example, when playing Spotify material.

When you update to 4.2, you'll also need to update NQPTP to version 1.2.

**Comment by noelhibbard** [comment]
I'm getting this when connecting:
```0.000290199 "audio_pa.c:222" *fatal error: failed to connect to the pulseaudio context -- the error message is "Connection refused".```

Older builds work fine. I feel like this has happened before in the past where some code changes were not applied to the PA backend. I can't find my past report of this though.

**Comment by mikebrady** [comment]
Thanks for the report. I’ll take a look. Was there anything special about the environment? E.g. is it in a normal Linux environment or a Docker environment?

**Reply by noelhibbard** [reply]
I clone the shairport-sync repo and then modify the Docker file with sed to add curl and all the PA dependencies and then do a custom build. I need to look into the official PA build but haven't had a chance to look.

**Reply by mikebrady** [reply]
Many thanks for that. I'll do some digging...

**Comment by mikebrady** [comment]
Okay, so I've installed Shairport Sync and NQPTP in a Ubuntu 18.04 and a Ubuntu 20.04 and it works. The configuration command was:
```
$ ./configure --sysconfdir=/etc --with-pa --with-soxr --with-avahi --with-ssl=openssl --with-airplay-2
```
So maybe there's something about the Docker image.

**Reply by mikebrady** [reply]
Just posted a question at [ci: build docker with pa](https://github.com/mikebrady/shairport-sync/pull/1642).

**Reply by noelhibbard** [reply]
> Okay, so I've installed Shairport Sync and NQPTP in a Ubuntu 18.04 and a Ubuntu 20.04 and it works. The configuration command was:
> 
> ```
> $ ./configure --sysconfdir=/etc --with-pa --with-soxr --with-avahi --with-ssl=openssl --with-airplay-2
> ```
> 
> So maybe there's something about the Docker image.

Okay, I'll do some more digging. I noticed pulse is part of the normal docker image now so I need to make some changes on my end to just use the standard image so at least we are closer to apples and apples.

I will report back what I find.

**Reply by mikebrady** [reply]
Many thanks, @noelhibbard.

**Reply by noelhibbard** [reply]
I thought I responded to this but I don't see my response. Maybe I typed it but never actually posted it. The official docker image had the same issue. The solution was to set the server in the PA section of the shairport-sync.conf file. In the past sps would find the server using the PULSE_SERVER environment value so it didn't need to be set in the config file.

I suspect this isn't actually an sps thing but maybe something changed in the PA libraries that are in the docker image.

**Reply by noelhibbard** [reply]
Okay, digging some more, within the docker container if you open ```/etc/pulse/client.conf``` you will see this line:
```
default-server = unix:/tmp/pulseaudio.socket
```

That must be what was over writing my PULSE_SERVER environment variable. So yeah, it looks like my issue had nothing to do with SPS but a change to the environment in the newer docker image.

**Reply by mikebrady** [reply]
Thanks. So perhaps this could be associated with the transition from Alpine Linux from 3.12 to 3.17?

Does it mean that you've gotten the official Docker image working the way you'd expect?

**Reply by noelhibbard** [reply]
> Thanks. So perhaps this could be associated with the transition from Alpine Linux from 3.12 to 3.17?
> 
> Does it mean that you've gotten the official Docker image working the way you'd expect?

Actually no, I think it was broken for me when PA support was added to the official image which added ```/etc/pulse/client.conf``` which was overwriting my PULSE_SERVER environment variable.

I removed the PULSE_SERVER env var from my docker-compose and then added a volume mount that looks like this:
```yml
volumes:
  - /var/run/pulse/native:/tmp/pulseaudio.socket
```

This fixes the issue without having to modify all my shairport-sync.conf files or make any changes to the official docker image.

I also updated/streamlined my instructions over [here](https://github.com/noelhibbard/shairport-sync-docker) to reflect these changes.
